trino/client.py

```diff
@@ -234,6 +236,15 @@ def timezone(self):
         with self._object_lock:
             return self._timezone
 
+    def _format_roles(self, roles):
+        formatted_roles = {}
+        for catalog, role in roles.items():
+            if role in ("NONE", "ALL") or ROLE_PATTERN.match(role) is not None:
```

Comment:
If role matches the ROLE_PATTERN can we use `warnings.warn` to tell user to fix it instead of silently accepting it?

Long term supporting both syntaxes may not actually solve the confusion.

@mdesmet Can you do this as a follow-up PR?